Update development dependencies

watsonbox opened the pull request.

godric commented on spec/checker_jobs/notifiers/bugsnag_spec.rb:
```diff
@@ -15,42 +15,38 @@
     # rubocop:disable RSpec/ExampleLength
     it "is an Error and have an explicit message", aggregate_failures: true do
       # We hook deep into Bugsnag to ensure we send the right payload.
       # This ensure we're only compatible with Bugsnag 5 and not Bugsnag 6.
```
Did the changes bellow invalid what is said in this comment ?
In either case shouldn't it be reflected in the `gemspec` version requirements ?

godric replied:
👋

watsonbox replied:
Sorry yes I did miss it. Well, nothing in the runtime code changed between these major versions, it's just the specs. So technically you could work with Sidekiq 5 in development just fine, though the specs would fail.

Since it's only development deps, I decided to do as you suggest and adapt the `gemspec`.

godric reviewed (changes requested):
It'd be nice to wait until after #191 is merged so we can rebase this and get some CI support 🙏

godric commented on .rubocop.yml:
```diff
@@ -1,7 +1,7 @@
 
 AllCops:
-  TargetRubyVersion: 2.3
+  TargetRubyVersion: 2.6
```
This is automatic from the `.ruby-version` and should probably be left out.

godric reviewed (changes requested):
I think you missed by last comment, I added a new one about the redundant `TargetRubyVersion`  in `rubocop.yml`

watsonbox commented:
> I think you missed by last comment, I added a new one about the redundant `TargetRubyVersion` in `rubocop.yml`

Indeed, I updated that and thanks also for the simplifaction of `rubocop.yml`.

watsonbox commented:
Third time's the charm 🤞

godric reviewed (approved):
LGTM!
